beacon-chain/core/transition/transition.go

```diff
@@ -211,18 +211,23 @@ func ProcessSlots(ctx context.Context, state state.BeaconState, slot primitives.
 	if err != nil {
 		return nil, err
 	}
-
-	if cachedState != nil && !cachedState.IsNil() && cachedState.Slot() < slot {
-		highestSlot = cachedState.Slot()
+	cachedSlot := primitives.Slot(0)
+	if cachedState != nil {
+		cachedSlot = cachedState.Slot()
+	}
+	if cachedState != nil && !cachedState.IsNil() && cachedSlot <= slot && cachedSlot > highestSlot {
```

Comment:
Probably best to extract long `cachedState != nil && !cachedState.IsNil() && cachedSlot <= slot && cachedSlot > highestSlot {` into a helper function

Reply:
would rather not cause I want to keep the return of `cachedSlot` and this would force the function signature to return something which makes this awkward.